galpy/potential_src/potential_c_ext/SpiralArmsPotential.c

```diff
@@ -0,0 +1,763 @@
+        double dBn_dR = dB_dR(R, H, n, N, sin_alpha);
+        double dDn_dR = dD_dR(R, H, n, N, sin_alpha);
+
+        double cos_ng = cos(n * g);
+        double sin_ng = sin(n * g);
+
+        double zKB = z * Kn / Bn;
+        double sechzKB = 1 / cosh(zKB);
+
+        sum += Cn * pow(sechzKB, Bn) / Dn * ((n * dg_dR / Kn * sin_ng
+                                              + cos_ng * (z * tanh(zKB) * (dKn_dR / Kn - dBn_dR / Bn)
+                                                          - dBn_dR / Kn * log(sechzKB)
+                                                          + dKn_dR / Kn / Kn
+                                                          + dDn_dR / Dn / Kn))
+                                             + cos_ng / Kn / Rs);
+    }
+
+    return -amp * H * exp(-(R - r_ref) / Rs) * sum;
+}
+
+
+double SpiralArmsPotentialzforce(double R, double z, double phi, double t,
+                                 struct potentialArg *potentialArgs) {
+
+    // Get args
+    double *args = potentialArgs->args;
+
+    int nCs = (int) *args++;
+    double amp = *args++;
+    double N = *args++;
+    double sin_alpha = *args++;
+    double tan_alpha = *args++;
+    double r_ref = *args++;
+    double phi_ref = *args++;
+    double Rs = *args++;
+    double H = *args++;
+    double omega = *args++;
+
+    phi = phi - omega * t;
+
+    double g = gam(R, phi, N, phi_ref, r_ref, tan_alpha);
+
+    // Return the zforce (-dPhi / dz)
+    double sum = 0;
+    int n;
+
+    for (n = 1; n <= nCs; n++) {
+        double Cn = *args++;
+        double Kn = K(R, n, N, sin_alpha);
+        double Bn = B(R, H, n, N, sin_alpha);
+        double Dn = D(R, H, n, N, sin_alpha);
+
+        double zKn_Bn = z * Kn / Bn;
+
+        sum += Cn / Dn * cos(n * g) * tanh(zKn_Bn) / pow(cosh(zKn_Bn), Bn);
+    }
+
+    return -amp * H * exp(-(R - r_ref) / Rs) * sum;
+}
+
+double SpiralArmsPotentialphiforce(double R, double z, double phi, double t,
+                                   struct potentialArg *potentialArgs) {
+
+    // Get args
+    double *args = potentialArgs->args;
+
+    int nCs = (int) *args++;
+    double amp = *args++;
+    double N = *args++;
+    double sin_alpha = *args++;
+    double tan_alpha = *args++;
+    double r_ref = *args++;
+    double phi_ref = *args++;
+    double Rs = *args++;
+    double H = *args++;
+    double omega = *args++;
+
+    phi = phi - omega * t;
+
+    double g = gam(R, phi, N, phi_ref, r_ref, tan_alpha);
+
+    // Return the phiforce (-dPhi / dphi)
+    double sum = 0;
+    int n;
+
+    for (n = 1; n <= nCs; n++) {
+        double Cn = *args++;
+        double Kn = K(R, n, N, sin_alpha);
+        double Bn = B(R, H, n, N, sin_alpha);
+        double Dn = D(R, H, n, N, sin_alpha);
+
+        sum += N * n * Cn / Dn / Kn / pow(cosh(z * Kn / Bn), Bn) * sin(n * g);
+    }
+
+    return -amp * H * exp(-(R - r_ref) / Rs) * sum;
+}
+
+double SpiralArmsPotentialR2deriv(double R, double z, double phi, double t,
```

Comment:
Again, great that you added this function, because it may becomes useful in the future, but currently galpy does not use the general forms of the second derivatives in C (only the planar ones), so add
```
//LCOV_EXCL_START
```
 before this function.